server/src/main/java/io/crate/expression/reference/doc/lucene/DocCollectorExpression.java

```diff
@@ -56,13 +58,28 @@ public final void setNextReader(ReaderContext context) throws IOException {
         this.context = context;
     }
 
-    public static LuceneCollectorExpression<?> create(final Reference reference) {
+    public static LuceneCollectorExpression<?> create(final Reference reference,
+                                                      Predicate<Reference> isParentReferenceIgnored) {
         assert reference.column().name().equals(SysColumns.DOC.name()) :
             "column name must be " + SysColumns.DOC.name();
         if (reference.column().isRoot()) {
             return new RootDocCollectorExpression(reference);
         }
-        return new ChildDocCollectorExpression(reference);
+
+        Function<Object, Object> valueConverter;
+        if (isParentReferenceIgnored.test(reference)) {
+            valueConverter = val -> {
+                try {
+                    return reference.valueType().sanitizeValue(val);
+                } catch (ClassCastException | IllegalArgumentException e) {
+                    return null;
+                }
+            };
```

Comment:
I'm a bit confused why we now have both this try/catch with `sanitizeValue` but also `sanitizeValueFromTranslog`. 
Why do we need both approaches?

Reply:
Good point, I'll rename the method an use it here as well.